warehouse/upload.go

```diff
@@ -1622,6 +1624,50 @@ func (job *UploadJobT) getLoadFilesTableMap() (loadFilesMap map[tableNameT]bool,
 	return
 }
 
+func (job *UploadJobT) destinationRevisionIDMap() (revisionIDMap map[string]backendconfig.DestinationT, err error) {
+	revisionIDMap = make(map[string]backendconfig.DestinationT)
+	revisionIDs, err := distinctDestinationRevisionIdsFromStagingFiles(struct {
+		sourceID           string
+		destinationID      string
+		startStagingFileID int64
+		endStagingFileID   int64
+	}{sourceID: job.warehouse.Source.ID, destinationID: job.warehouse.Destination.ID, startStagingFileID: job.upload.StartStagingFileID, endStagingFileID: job.upload.EndStagingFileID})
```

Comment:
Can we decouple and create the variable beforehand for better readability ?

Reply:
Done.